### ISSUE TITLE
Atlas measurements

### ISSUE THREAD

**Issue by romain-fontugne** [InternetHealthReport/internet-yellow-pages#81]
Get the currently running RIPE Atlas measurement (https://atlas.ripe.net/api/v2/measurements/) with the list of probes.

Add the following relationships:
```cypher
(:AtlasProbe)-[:PART_OF]->(:AtlasMeasurement)-[:TARGET]->(:DomainName)
```
and
```cypher
(:AtlasProbe)-[:PART_OF]->(:AtlasMeasurement)-[:TARGET]->(:IP)
```

**Comment by mohamedawnallah** [comment]
I’m gonna work on this issue :)

**Comment by m-appel** [comment]
Thanks! To spare you some work, this is the actual API request, Romain posted only the light version:

https://atlas.ripe.net/api/v2/measurements/?is_public=true&status=2&optional_fields=current_probes

This returns only active measurements and includes the ids of the participating probes. See also [the reference.](https://atlas.ripe.net/docs/apis/rest-api-reference/#measurements)

**Comment by mohamedawnallah** [comment]
We're currently dealing with four main entities within our dataset:

- `target`: Typically an IP Address or a domain address.
- `target_asn`: Contains both null and valid ASes, which could be useful in queries.
- `target_ip`: Some data points may have null entries.
- `target_prefix`: Might contain valid target Prefixes.

Two main questions arise:

1) Should relationships involving the `AtlasMeasurement` entity extend beyond `IP` and `DomainName` entities to include `Prefix` and `AS` entities?

2) When encountering an invalid domain name, similar to the one shown in the screenshot, how should we address this? Should we handle the situation atomically by discarding the entire data point (including the relationship with `IP`) if the `DomainName` is invalid? Or if there is an internal tool that converts an IP address to a domain name similar to [ip-to-hostname tool ](https://dnschecker.org/ip-to-hostname.php?query=82.197.70.19) ? Alternatively, is it acceptable for an `AtlasMeasurement` node to have a relationship with `IP` but not with a valid `DomainName`, allowing for a more graceful resolution of this discrepancy?

The screenshot provides an overview of RIPE Atlas measurements, the specific data points accessible here: [RIPE Atlas Measurements](https://atlas.ripe.net/api/v2/measurements/?is_public=true&optional_fields=current_probes&page=6&status=2)

![2023-12-13_15-39](https://github.com/InternetHealthReport/internet-yellow-pages/assets/69568555/859159c6-5372-4667-9cb2-8ffa834cfa12)

**Comment by m-appel** [comment]
With RIPE Atlas you can target either an IP directly, or specify a hostname that is resolved to one (or more) IPs by the system. So `target` can indeed be either of these cases, like you mentioned.

Both are valid, so in the example above you would only create a link to `IP` without any domain name. Btw. there is also a `resolved_ips` list that might be preferred over the `target_ip` field. I believe this can contain multiple IPs if a domain name was specified as the target, but I'm not sure. In this case we want to add links to all of them.

Long story short: Always add `target_ip` (or `resolved_ips`) and add `target` if it's a domain name. We do not need `target_asn` and `target_prefix` for now, since we have this connection already in our database (and technically the measurement is not really targeting an AS or prefix, but a specific host within them).
